src/dissectors/ec_o5logon.c

```diff
@@ -1,43 +1,62 @@
 /*
-    ettercap -- dissector for Oracle O5LOGON protocol -- TCP 1521
-
-    Copyright (C) Dhiru Kholia (dhiru at openwall.com)
-
-    Tested with Oracle 11gR1 and 11gR2 64-bit server and Linux +
-    Windows SQL*Plus clients.
-
-    It does work with Nmap generated packets though the code is hacky.
-
-    This program is free software; you can redistribute it and/or modify
-    it under the terms of the GNU General Public License as published by
-    the Free Software Foundation; either version 2 of the License, or
-    (at your option) any later version.
-
-    This program is distributed in the hope that it will be useful,
-    but WITHOUT ANY WARRANTY; without even the implied warranty of
-    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-    GNU General Public License for more details.
-
-    You should have received a copy of the GNU General Public License along
-    with this program; if not, write to the Free Software Foundation, Inc.,
-    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
-
-*/
+ * ettercap -- dissector for Oracle O5LOGON protocol -- TCP 1521
+ *
+ * Copyright (c) 2012-2014 Dhiru Kholia (dhiru at openwall.com)
+ * Copyright (c) 2016 magnum
+ *
+ * Tested with Oracle 11gR1 and 11gR2 64-bit server and Linux +
+ * Windows SQL*Plus clients. Now also works with Oracle 12. That
+ * version avoids the known-plain vulnerability from PKCS#7 padding
+ * so more fields are needed for a full attack.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ */
 
 #include <ec.h>
 #include <ec_decode.h>
 #include <ec_dissect.h>
 #include <ec_session.h>
 
+//#define O5DEBUG
```

Comment:
I'd rather recommend defining this (prefixed with EC_) with the value 0 for off and 1 for on.
to include such a block you the do `#if EC_O5DEBUG` instead of `#ifdef ...`.
But this is really somewhat cosmetic and flavor.